io/disk/Avago_storage_adapter/avago3008.py

```diff
@@ -40,47 +40,73 @@ def setUp(self):
+        if self.raidlevel == 'raid1':
+            self.raid_disk = " ".join(self.disk[0:2:1]).strip(" ")
+        if self.raidlevel == 'raid1e':
+            if self.number_of_disk < 3:
+                self.skip("Raid1e needs minimum of 3 disks to be tested")
+            else:
+                self.raid_disk = " ".join(self.disk).strip(" ")
+        if self.raidlevel == 'raid10':
+            if self.number_of_disk < 4:
+                self.skip("Raid10 needs minimum of 4 disks to be tested")
+            else:
+                self.raid_disk = " ".join(self.disk).strip(" ")
+        if self.raidlevel == 'raid0':
+            self.raid_disk = " ".join(self.disk).strip(" ")
 
-    def test_adapterlist(self):
+    def test_run(self):
 
         """
-        Lists all the LSI adapters attached to the mahcine
-        :return:
-        """
-        cmd = "./sas3ircu list"
-        if process.system(cmd, ignore_status=True, shell=True) != 0:
-            self.fail("Failed to list all the Avogo adapters")
-
-    def test_adapterdetails(self):
-
-        """
-        Display controller, volume and physical device info
+        Decides which functions to run for given raid_level
         """
 
-        cmd = "./sas3ircu %d display" % self.controller
+        cmd = "echo -e 'YES\nNO' | ./sas3ircu %d delete" \
+              % (self.controller)
         if process.system(cmd, ignore_status=True, shell=True) != 0:
-            self.fail("Failed to display details of drives and VR vloumes")
+            self.fail("Unable to clear entire configuration before starting")
+        if self.raidlevel == 'raid0':
+            self.basictest()
+        else:
+            self.extensivetest()
 
-    def test_createraid(self):
+    def extensivetest(self):
 
         """
-        This function creates raid1 array
+        Lists all the LSI adapters attached to the mahcine
+        :return:
         """
 
-        if self.number_of_disk >= 2:
-            cmd = "./sas3ircu %d create RAID1 max %s %s vr1 noprompt" \
-                 % (self.controller, self.disk[0], self.disk[1])
+        self.adapterlist()
+        self.adapterdetails()
+        self.createraid()
+        self.adapterdetails()
+        self.adapter_status()
+        self.set_online_offline("offline")
+        self.set_online_offline("online")
+        for _ in range(0, 5):
+            for state in ['offline', 'online']:
+                self.set_online_offline(state)
+                time.sleep(10)
+        if self.spare:
+            cmd = "echo -e 'YES\nNO' | ./sas3ircu %d hotspare %s" \
+                 % (self.controller, self.spare)
             if process.system(cmd, ignore_status=True, shell=True) != 0:
-                self.fail("Failed to create RAID1 on the drives")
+                self.fail("Failed to set hotspare drive")
         else:
-            self.skip("Please provide minimum of two drives")
+            self.fail("Hot-spare disk not specified")
```

Comment:
@narasimhan-v  I have planned few more test cases for this script, I will definitely consider this suggestion, in coming patches.